### PR TITLE
Add sample kubernetes deployment

### PR DESCRIPTION
**Pull request by natalysheinin**
Adding a Kubernetes Deployment manifest that can be used to deploy falcosidekick to a Kubernetes environment.

Verified deployment works on Minikube.
image

cc @Issif

### CODE REVIEW

**Comment by Issif** [comment]
Thanks a lot for this contribution. I'm working on a local branch about same topic for a while, just didn't had enough time to move forward (I'm still bad with helm formalism, so any help is welcome).

I would like a solution without modifying the chart for adapting, I prefer to use configmap.yml + values.yml, especially because I'm planning a version 2.0.0 with a lot of changes. Environment variables are useful but we could do better with a hierarchy of configuration methods "env vars > yaml config file > default values", I will surely using Viper package for doing that : https://github.com/spf13/viper.

My point is to provide several methods for configuration and possible overridings.

You choose to create a `Service` with a `ServiceType` `NodePort` but I prefer `ClusterIP`, access from outside the cluster should be prohibited. I'm asking myself too if we should use a classic deployment with `ReplicaSet` or a `DaemonSet`.

I think I will take ideas from https://github.com/helm/charts/tree/master/stable/traefik, traefik is piece of software I know well and I really like, its mechanism for handling configuration inspires me (will use yaml instead of toml to be consistent with `falco.yaml`).

Regards,

**Comment by natalysheinin** [comment]
@Issif do you need me to adjust anything here for it to get merged?

File: kubernetes/falco.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This is not the full falco.yaml just the program_output config
+program_output:
+  enabled: true
+  keep_alive: false
+  program: "curl -d @- falcosidekick.sysdig-staging.svc.cluster.local:2801/"
```
Comment (Issif) [inline kubernetes/falco.yaml]:
```suggestion
  program: "curl -d @- falcosidekick.falco.svc.cluster.local:2801/"
```

I think we should add falcosidekick in same namespace than falco

File: kubernetes/falcosidekick-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: falcosidekick
+    spec:
+      serviceAccount: falcosidekick
+      containers:
+        - name: falcosidekick
+          image: issif/falcosidekick:1.1.0
```
Comment (Issif) [inline kubernetes/falcosidekick-deployment.yaml]:
```suggestion
          image: falco/falcosidekick:2.7.2
```

falcosidekick is now an official project, we moved it's repository

**Comment by Issif** [comment]
@natalysheinin 	Yes please :

- I added a lot of new outputs, please consider to add them
- I reviewed your last commit, please take a look at my comments
- We added helm chart in `/deploy/helm` please move your yaml into `/deploy/kubernetes`
- Add a NOTES.txt with instructions about how to use your yaml with kubectl (creation of secrets before for example)

**Comment by Issif** [comment]
I'm also thinking about using a ConfigMap (#33), for direct yaml and helm chart

File: kubernetes/falcosidekick-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          imagePullPolicy: Always
+          ports:
+          - containerPort: 2081
```
Comment (DannyPat44) [inline kubernetes/falcosidekick-deployment.yaml]: Typo 2801

File: kubernetes/falcosidekick-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            name: svc-port
+          env:
+          # CREATE THE K8S SECRETS BEFORE UNCOMMENTING
+          # https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret-using-kubectl-create-secret
+          # Example: kubectl create secret falcosidekick-slack-secret --from-file=./slack-secret.txt 
+          # - name: SLACK_WEBHOOK_URL
```
Comment (DannyPat44) [inline kubernetes/falcosidekick-deployment.yaml]: This should be "SLACK_WEBHOOKURL" no?

File: kubernetes/falcosidekick-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          #       key: datadog_api_key
+          - name: SLACK_FOOTER
+            value: ""
+          - name: SLACK_ICON
+            value: ""
+          - name: SLACK_OUTPUT_FORMAT
+            value: "all"
+          - name: DEBUG
+            value: "true"
+          args: [ "/falcosidekick"]
```
Comment (DannyPat44) [inline kubernetes/falcosidekick-deployment.yaml]: This appears not to be needed.

File: kubernetes/falcosidekick-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+          # - name: DATADOG_API_KEY
+          #   valueFrom:
+          #     secretKeyRef:
+          #       name: falcosidekick-datadog-secret
+          #       key: datadog_api_key
+          - name: SLACK_FOOTER
+            value: ""
+          - name: SLACK_ICON
+            value: ""
+          - name: SLACK_OUTPUT_FORMAT
```
Comment (DannyPat44) [inline kubernetes/falcosidekick-deployment.yaml]: This should be SLACK_OUTPUTFORMAT

**Pull request closed without merging**